app/models/goal.py

```
@@ -1,5 +1,35 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import String
 from ..db import db
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column(String(100), nullable=False)
+    
+    tasks = relationship("Task", back_populates="goal", cascade="all, delete-orphan")
+    
+    @classmethod
+    def from_dict(cls, goal_data):
+        """
+        Creates a new Goal instance from a dictionary
+        """
+        if not goal_data.get("title"):
+            return None
```

We can handle bad information a number of ways, but this is a place where it would be totally valid and normal to let the exception raise if someone provides bad data to the request, and use a try/except in the route to catch the issue and respond accordingly. 

This feedback applies to `Task.from_dict` as well